app/helpers/projects_helper.rb

```diff
@@ -94,6 +95,17 @@ def more_badges_link
     end
   end
 
+  def populate_project_from_forge(url, api)
+    match = Forge::Match.first(url)
+    Timeout.timeout(Forge::Match::MAX_FORGE_COMM_TIME) { @project = match.project } if match
+  rescue Timeout::Error, OpenURI::HTTPError, URI::InvalidURIError
+    if api == true
+      render json: t('.forge_time_out', name: match.forge.name)
+    else
+      flash.now[:notice] = t('.forge_time_out', name: match.forge.name)
+    end
+  end
```

Comment:
Setting instance variables for an object in one class and then consuming it in another is a bad code pattern. 
See [this](https://softwareengineering.stackexchange.com/a/214853) for more detailed explanation.
To keep it short, try passing local variables when possible. Instance variables seem clean but lead to unmanageable mess in the future. As a rule, using them in the same class is usually fine. Passing variables around seems ugly but it is better practice.

Reply:
In the above code, pass project instead of setting `@project`. Also for the conditional, check if the following works:
```
   api ? render json: t('.forge_time_out', name: match.forge.name)
    : flash.now[:notice] = t('.forge_time_out', name: match.forge.name)
```
Not sure if rubocop will allow it.

Reply:
I can change this one but it is having a dependency on UI and API. It might take time to work on this. After the change, I have to test everything in UI and API.

Can we skip this one for this PR?

I will take care on next PR.